CONTRIBUTING.md

```diff
@@ -1,59 +1,602 @@
+
+Our GitHub Actions integration will run the automated tests and give you feedback in the review section. We will review your
+changes and give you feedback as soon as possible.
+
+We recommend that you regularly rebase from `main` of the original repo to keep your
+branch up to date.
+
+Please ensure that `Allow edits and access to secrets by maintainers` is checked.
+While the maintainers will of course wait for you to edit your own work, if you are
+unresponsive for over a week, they may add corrections or even complete the work for you,
+especially if what you are contributing is very cool :metal: .
+
+PRs which adhere to our guidelines are more likely to be accepted
+(when opening the PR, please use the checklist in the template):
+
+1. **The description is thorough.** When writing your description, please be as detailed as possible: don't make people
+  guess what you did or simply link back to the issue (the issue explains the problem
+  you are trying to solve, not how you solved it.)
+  Guide your reviewers through your solution by highlighting
+  key changes and implementation choices. Try and pre-empt any obvious questions
+  they may have. Providing snippets (or screenshots) of output is very helpful to
+  demonstrate new behaviour or UX changes. (Snippets are more searchable than screenshots,
+  but we wont be mad at a sneak peek at your terminal envs :eyes: .)
+
+1. **The change has been manually tested.** If you are supplying output above
+  then that can be your manual test, with proof :clap: .
+
+1. **The PR has a snappy title**. Your PR title will end up in the release notes,
+  so make it a good one. Using the same rule as for the title of a [commit message](#commit-message-formatting)
+  is generally a good idea. Try to use the [imperative](https://en.wikipedia.org/wiki/Imperative_mood) and centre it around the behaviour
+  or the user value it delivers, rather than any implementation detail.
+
+    eg: `"changed SomeFunc in file.go to also do thing X"`
+    is not useful for folks stopping by to quickly see what new stuff they can do with
+    `flintlock`, save that for commit messages or the PR description.
+    The title `"(feat): Add support for XYZ"` conveys the intent concisely and clearly.
+
+1. **There are new tests for new code.** This is very much required, see [testing standards](#testing-standards)
+  below.
+
+1. **There are new tests for old code!** This will earn you the title of "Best Loved
+  and Respected Contributor" :boom: :sunglasses: .
+
+1. **There are well-written commit messages** ([see below](#commit-message-formatting))
+  which will make future debugging fun.
+
+
+In general, we will merge a PR once a maintainer has reviewed and approved it.
+Trivial changes (e.g., corrections to spelling) may get waved through.
+For substantial changes, more people may become involved, and you might get asked to resubmit the PR or divide the
+changes into more than one PR.
+
+### Testing standards
+
+TODO
```

Comment:
Is it in progress, or a mark to do it later in a separate PR? Do we need this section if it's only a mark?

Reply:
ah yeh technically a mark, i want to have a discussion about it first